p2p/tests/sessions.rs

```diff
@@ -568,3 +568,50 @@ fn p2p_sessions_register_more_than_limit() {
     assert_eq!(sessions.get_num_outbound_sessions(), 3);
     assert_eq!(sessions.num_missing_outbound(), 0);
 }
+
+#[test]
+/// Check that peer addresses are considered "similar" if the first two octets of the IP are
+/// matching.
+fn p2p_peer_address_is_similar_to_inbound_session() {
```

Comment:
Can we have a similar test that checks that after calling unregister_session, the ip range is available again?